### PR TITLE
Sachhidanand bit array

### PR DESCRIPTION
**Pull request by sachhidanandjha**

### CODE REVIEW

File: Sachhidanand/bit_array/Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+//*****Makefile for bit_array*****
```
Comment (aloknigam247) [inline Sachhidanand/bit_array/Makefile:1]: Makefile comments starts with #

File: Sachhidanand/bit_array/Makefile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+bit_array: main.o b_array.o b_array.h
```
Comment (aloknigam247) [inline Sachhidanand/bit_array/Makefile:3]: output name should be demo
array.h is not needed here

File: Sachhidanand/bit_array/definition.c [hunk 3]
```diff
@@ -0,0 +1,18 @@
+//Definition of functions
+
+void b_create(int size_t){
```
Comment (aloknigam247) [inline Sachhidanand/bit_array/definition.c:3]: size_t is the type and not the variable name,
```suggestion
void b_create(size_t size){
```

File: Sachhidanand/bit_array/definition.c [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    int *array;
```
Comment (aloknigam247) [inline Sachhidanand/bit_array/definition.c:5]: is int correct here, what if my number of bits is 1. then how many bits will we waste. should we take another type ?

File: Sachhidanand/bit_array/definition.c [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+    if (array==NULL)
+    {
+        printf("Error! Try Again");
+    }
+    
+    for(int i=0; i<(sizeof(array)/sizeof(array+0)); i++)
+    {
+        scanf("%d",*(array+i));
+        }
```
Comment (aloknigam247) [inline Sachhidanand/bit_array/definition.c:17]: we do not want to take input from user here

**Review by aloknigam247 — CHANGES_REQUESTED** [review]

File: Sachhidanand/bit_array/definition.c (hunk 4 above)
Comment (sachhidanandjha) [reply Sachhidanand/bit_array/definition.c:5]: should we take unsigned char then?

File: Sachhidanand/bit_array/definition.c (hunk 3 above)
Comment (sachhidanandjha) [reply Sachhidanand/bit_array/definition.c:3]: okay sir

File: Sachhidanand/bit_array/definition.c [hunk 6]
```diff
@@ -0,0 +1,43 @@
+//Definition of functions
+typedef struct b_array
+{
+    unsigned char *array;
+    unsigned int bit;
+}b_array;
```
Comment (aloknigam247) [inline Sachhidanand/bit_array/definition.c:6]:
```suggestion
typedef struct
{
    unsigned char *array;
    unsigned int size;
} barray;
```

File: Sachhidanand/bit_array/b_array.c [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    }  
+}
+
+void b_destroy(b_array *arr){
+
+    if (arr->array!=NULL)
+        free(arr->array);   
+    if (arr!=NULL)      
+        free(arr);
```
Comment (aloknigam247) [inline Sachhidanand/bit_array/b_array.c:52]: what if user by mistake calls b_destroy again with same pointer ?

File: Sachhidanand/bit_array/b_array.c [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+}
+void b_destroy(b_array *arr){
+
+   
+    if (arr==NULL)
+    {
+        printf("\nThis is Empty");
+        return;
+    }
+    if(arr)
```
Comment (aloknigam247) [inline Sachhidanand/bit_array/b_array.c:54]: do we need this if ? If line 49 is false then line 54 is obviously true

**Pull request merged**